packages/js/client/src/__tests__/env.spec.ts

```diff
@@ -0,0 +1,364 @@
+import {
+  buildAndDeployApi,
+  initTestEnvironment,
+  stopTestEnvironment,
+} from "@web3api/test-env-js";
+import { createWeb3ApiClient, Web3ApiClientConfig } from "..";
+import { Client, Plugin, PluginModules } from "@web3api/core-js";
+import { GetPathToTestApis } from "@web3api/test-cases";
+
+jest.setTimeout(200000);
+
+describe("Web3ApiClient", () => {
+  let ipfsProvider: string;
+  let ethProvider: string;
+  let ensAddress: string;
+
+  beforeAll(async () => {
+    const { ipfs, ethereum, ensAddress: ens } = await initTestEnvironment();
+    ipfsProvider = ipfs;
+    ethProvider = ethereum;
+    ensAddress = ens;
+  });
+
+  afterAll(async () => {
+    await stopTestEnvironment();
+  });
+
+  const getClient = async (config?: Partial<Web3ApiClientConfig>) => {
+    return createWeb3ApiClient(
+      {
+        ethereum: {
+          networks: {
+            testnet: {
+              provider: ethProvider,
+            },
+          },
+        },
+        ipfs: { provider: ipfsProvider },
+        ens: {
+          addresses: {
+            testnet: ensAddress,
+          },
+        },
+      },
+      config
+    );
+  };
+
+  const mockEnvPlugin = () => {
+    class MockEnvPlugin extends Plugin {
+      getModules(_client: Client): PluginModules {
+        return {
+          query: {
+            sanitizeEnv: async (env: { arg1: string }) => {
+              return { arg1: parseInt(env.arg1) };
+            },
+            queryEnv: () => {
+              return this.getEnv("query");
+            },
+          },
+          mutation: {
+            sanitizeEnv: async (env: { arg1: number }) => {
+              return { arg1: env.arg1.toString() };
+            },
+            mutationEnv: () => {
+              return this.getEnv("mutation");
+            },
+          },
+        };
+      }
+    }
+    return {
+      factory: () => new MockEnvPlugin(),
+      manifest: {
+        schema: ``,
+        implements: [],
+      },
+    };
+  };
+
+  describe("env types", () => {
```

Comment:
The only tests we're running here are "happy paths", and we're not testing the error cases that this PR tries to address.

I would request that we add tests for the following cases:
- No env provided, when one is required
- Incorrect env provided
- Env provided, when wrapper does not ask for one (shouldn't throw an error)

Reply:
I agree